### ISSUE TITLE
NYU classification: What modification do you do to your network ?

### ISSUE THREAD

**Issue by Scienceseb** [pmorerio/admd#7]
Hi, 
**for the NYU classification task using only RGB images.** 
I want to know if you have modified the resnet 50, what normalization do you used (im pretty sure its the one of ImageNet) and what kind of data augmentation do you used during training? In Pytorch I’m not able to get closed to your results (with RGB for exemple I get an accuracy of 45%), I’m using a resnet 50 pretrained on image net, in training I applied the normalization of image net and I used data augmentation (random horizontal flip and random crop).

I tried to find the information in your code, but not being used to Tensorflow I'm really lost.

Thank you for your help!

**Comment by pmorerio** [comment]
Hi!

As we state in the paper, the adversarial distillation is eased with a bottleneck, which we implement on top of block4 of Resnet50. Thus we take the block4 feature map [here](https://github.com/pmorerio/admd/blob/754d5cc8d3aeae252a5c2cb16f66d5aac2d31de6/NYUD/model.py#L33) (7x7x2048) and we add a 7x7 convolution with 128 filters, with 'VALID' padding, meaning you only apply the filter once, and get a 1x1x128 vector. After that, a 1x1 convolution returns a vector of the correct length for classification. The bottleneck is implemented [here](https://github.com/pmorerio/admd/blob/754d5cc8d3aeae252a5c2cb16f66d5aac2d31de6/NYUD/model.py#L37).

No data augmentation is done. [Here](https://github.com/pmorerio/admd/blob/754d5cc8d3aeae252a5c2cb16f66d5aac2d31de6/NYUD/solver.py#L33) you can see how we load the data and only subtract [imagenet mean](https://github.com/pmorerio/admd/blob/754d5cc8d3aeae252a5c2cb16f66d5aac2d31de6/NYUD/solver.py#L36).

Hope this helps.

**Comment by Scienceseb** [comment]
Thanks a lot I will tried that in PyTorch. Also why this modification ? Did you tried the resnet 50 without any modification, was it bad ?

**Comment by pmorerio** [comment]
This modification was added in order to have a lower dimensional vector to be given to the discriminator, in order to ease the adversarial hallucination. There is a paragraph in the paper - page 6 - which explains it. I did not try Resnet50 without bottleck, but to be honest I guess 1 more convolution after 50 does not make much difference in terms of accuracy.

**Comment by Scienceseb** [comment]
Yep I think the same, maybe the loss function you used change everything, softmax cross entropy with logits dont exist in PyTorch.

**Comment by pmorerio** [comment]
Softmax cross entropy with logits is only a way for Tensorflow of unifying the softmax operation with the cross entropy in a single function. This way it is optimized better for numerical stability.

**Comment by Scienceseb** [comment]
yep I think it almost the same as the cross entropy loss in PyTorch...I really dont know why 7% lower than you, only in PyTorch because in Tensorflow I got your RGB score and even better. You dont use dropout or ?

**Comment by Scienceseb** [comment]
And what is your batch size (4 according to your code) with batch norma (even with a batch number that low) ?

**Comment by pmorerio** [comment]
No dropout is used.

**Comment by Scienceseb** [comment]
I really dont understand why I got poor performance like that, are you ok in PyTorch if so will you mind checking my main.py (its not that big), like I said I use a pretrained resnet 50 that’s it.  Thanks a lot for your help, I really want to make a good comparison with your paper but I cant because of that disparity...

**Comment by pmorerio** [comment]
Hi. I can give it a try. But honestly I think the disparity may be due to the fact that TF and PyTorch have slightly different implementations of ResNet50. Also pretrained models are surely different.

**Comment by Scienceseb** [comment]
I will convert your Tensorflow code to PyTorch to make a fair comparison.
